src/useAnimations.tsx

```diff
@@ -12,7 +12,7 @@ type Api = {
   }
 }
 
-export function useAnimations(clips: AnimationClip[], root?: React.MutableRefObject<Object3D>) {
+export function useAnimations(clips: AnimationClip[], root?: React.MutableRefObject<Object3D | undefined>) {
```

Comment:
I've had a play with this & I don't believe this is the full solution. This solution means I can pass
```ts
const group = useRef<THREE.Group>()

useAnimations(actions, group)
```

Only providing I **use** the ref in the application. Because if we don't, we're passing a ref that has `ref.current === undefined` which passes the check on l17 & throws errors at L31/2. Also some people will define `null` as the initial value of their ref so this would therefore fail typechecking:
```ts
const group = useRef<THREE.Group>(null)

useAnimations(actions, group)
```

So this needs a bit more thought.... I'll have a play & thought myself today - two heads better than one 💡

Reply:
Good catch, interested in any other solutions you see.

Reply:
hey @Code47X, ive done some investigation in to how `useRef` works with TS, take this example:

```ts
const groupRef = useRef<THREE.Group>()
```

This equates to `groupRef = THREE.Group | undefined` hence why your solution worked, however if I pass `null` like this:
```ts
const groupRef = useRef<THREE.Group>(null)
```
It'll probably throw an error, because `null` is not `THREE.Group | undefined` so we'd have to pass `<THREE.Group | null>` to the `useRef` generic. So i think the best solution is just adding `null` to the type you created! 

Then we want to check where we use something like `root.current` we're checking if it's defined // not null. To avoid any errors. Sorry if i just over explained things you already knew, thought it'd be good as a record, regardless.

Reply:
@joshuaellis Yup, makes sense to me. I'll throw in null to the ref type and make sure the ref has a `current` value before its used in the hook. I plan to make the changes we discussed sometime tonight! Thanks!